packages/local_auth/local_auth_windows/windows/local_auth_plugin.cpp

```diff
@@ -0,0 +1,223 @@
+
+void LocalAuthPlugin::HandleMethodCall(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  if (method_call.method_name().compare("authenticate") == 0) {
+    Authenticate(method_call, std::move(result));
+  } else if (method_call.method_name().compare("getAvailableBiometrics") == 0) {
+    GetAvailableBiometrics(std::move(result));
+  } else if (method_call.method_name().compare("isDeviceSupported") == 0) {
+    IsDeviceSupported(std::move(result));
+  } else {
+    result->NotImplemented();
+  }
+}
+
+// Starts authentication process
+winrt::fire_and_forget LocalAuthPlugin::Authenticate(
+    const MethodCall<EncodableValue>& method_call,
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  auto reasonW = Utf16FromUtf8(GetArgument<std::string>(
+      "localizedReason", method_call.arguments(), std::string()));
+
+  auto biometricOnly =
+      GetArgument<bool>("biometricOnly", method_call.arguments(), false);
+  if (biometricOnly) {
+    result->Error("biometricOnlyNotSupported",
+                  "Windows doesn't support the biometricOnly parameter.");
+    co_return;
+  }
+
+  auto ucvAvailability =
+      co_await user_consent_verifier_->CheckAvailabilityAsync();
+
+  if (ucvAvailability ==
+      Windows::Security::Credentials::UI::UserConsentVerifierAvailability::
+          DeviceNotPresent) {
+    result->Error("NoHardware", "No biometric hardware found");
+    co_return;
+  } else if (ucvAvailability ==
+             Windows::Security::Credentials::UI::
+                 UserConsentVerifierAvailability::NotConfiguredForUser) {
+    result->Error("NotEnrolled", "No biometrics enrolled on this device.");
+    co_return;
+  } else if (ucvAvailability !=
+             Windows::Security::Credentials::UI::
+                 UserConsentVerifierAvailability::Available) {
+    result->Error("NotAvailable", "Required security features not enabled");
+    co_return;
+  }
+
+  try {
+    auto consentResult =
+        co_await user_consent_verifier_->RequestVerificationForWindowAsync(
+            reasonW);
+
+    result->Success(EncodableValue(
+        consentResult == Windows::Security::Credentials::UI::
+                             UserConsentVerificationResult::Verified));
+  } catch (...) {
+    result->Success(EncodableValue(false));
+  }
+}
+
+// Returns biometric types available on device
+winrt::fire_and_forget LocalAuthPlugin::GetAvailableBiometrics(
+    std::unique_ptr<MethodResult<EncodableValue>> result) {
+  try {
+    flutter::EncodableList biometrics;
+    auto ucvAvailability =
+        co_await user_consent_verifier_->CheckAvailabilityAsync();
+    if (ucvAvailability == Windows::Security::Credentials::UI::
+                               UserConsentVerifierAvailability::Available) {
+      biometrics.push_back(EncodableValue("fingerprint"));
+      biometrics.push_back(EncodableValue("face"));
+      biometrics.push_back(EncodableValue("iris"));
```

Comment:
Looks like https://github.com/flutter/plugins/pull/4806#discussion_r814057427 is still not resolved.

Reply:
I'm now returning both weak and strong, but they act the same, just to increase compatibility, as Windows Hello doesn't support picking the auth security "level". Do you believe we should actually add the "unknown" option?

Reply:
What is the spectrum of possible types of authentication? We certainly shouldn't return a level that suggests stronger auth is available than actually is.

Consider, say, a banking application that wants to allow using local_auth-based authentication only if a sufficiently strong biometric option is available.

Reply:
From the documentation: "Microsoft Passport PIN, Windows Hello biometric, or fingerprint reader". This APIs is the recommended one for secure authentication on Windows 10+.
https://docs.microsoft.com/en-us/windows/uwp/security/microsoft-passport#2-what-is-windows-hello

It is important to notice that not all Windows 10 devices support Windows Hello, as it will only work if a TPM module is present, and the biometric credentials are properly configured. You can't only use a PIN. Here is a list of devices that support it:
https://www.windowscentral.com/complete-list-laptops-support-windows-hello

The list is long, and growing, but developers should be aware that this is not guaranteed, and should provide a different auth method, just as they should with the existing API exposed by this plugin on other platforms.

This **is** the API we recommend banking apps to use if they want a strong auth validation, as it is the same one that will be used if available during the Windows Sign-in process.

Reply:
Thanks, that's useful context.

Since we documented `strong` as:

```
  /// Any biometric (e.g. fingerprint, iris, or face) on the device that the
  /// platform API considers to be strong.
```

I'm fine with `weak` and `strong` if you think that description applies for Windows (which it sounds like is the case).

Reply:
I would say so, but it is important to notice that a PIN can still be used and will always be available (as long as Hello is enabled). This is mostly in case the user had an injury and can't use the biometric device, for example, but it is the user's choice, not the developers.

Reply:
That should be fine, since Android and iOS allow that as well with `biometricOnly` set to false.